cdap-data-fabric/src/main/java/co/cask/cdap/data2/dataset2/lib/partitioned/PartitionedFileSetDataset.java

```diff
@@ -316,14 +317,13 @@ public void addPartition(PartitionKey key, String path, Map<String, String> meta
   }
 
   private void addPartition(PartitionKey key, String path, Map<String, String> metadata, boolean filesCreated) {
-    AddPartitionOperation operation = new AddPartitionOperation(key, path, filesCreated);
-    operationsInThisTx.add(operation);
     byte[] rowKey = generateRowKey(key, partitioning);
     Row row = partitionsTable.get(rowKey);
     if (!row.isEmpty()) {
-      throw new DataSetException(String.format("Dataset '%s' already has a partition with the same key: %s",
```

Comment:
I thought about that, but we'd still need to generate the rowkey again.
Edit: I can make the `assertNotExists` return the rowkey, to avoid generating it twice, and still be able to reuase the new method `assertNotExists` here.